### PR TITLE
chore: update dependencies from webapp and server

### PR DESCRIPTION
**Pull request by marcos-cereijo-pexip**
Updated dependencies from webapp and server.

- In case of the server, checked the Mattermost template for plugins and adapted it to our case. Started using the new public API: https://github.com/mattermost/mattermost-plugin-starter-template/tree/master
- For the webapp, removed the "react-script" dependency, which was deprecated, and configured webpack and jest to work without it.
- Modified the code to work with the new eslint configuration.
- Had to maintain "react-tooltip: 4.5.1" for now, since the version versions 5.x.x crash and the same happens for "@mui/material" and "@pexip/components".

### CODE REVIEW

**Comment by marcos-cereijo-pexip** [comment]
I ran Snyk manually and found some issues related to one of the dependencies.

The dependency that we use is `github.com/mattermost/mattermost/server/public/plugin@0.0.12`, but this one use internally  `github.com/hashicorp/go-plugin` which has a problem with one of its internal dependencies and a license. @matthooper1 , do you know if we could ignore this issues for now?

Here is the list of issues:

```
✗ Medium severity vulnerability found in google.golang.org/protobuf/encoding/protojson
  Description: Stack-based Buffer Overflow
  Info: https://security.snyk.io/vuln/SNYK-GOLANG-GOOGLEGOLANGORGPROTOBUFENCODINGPROTOJSON-6137908
  Introduced through: github.com/mattermost/mattermost/server/public/plugin@0.0.12
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > google.golang.org/grpc/reflection@1.60.0 > google.golang.org/grpc/reflection/grpc_reflection_v1alpha@1.60.0 > google.golang.org/grpc@1.60.0 > google.golang.org/grpc/internal/transport@1.60.0 > google.golang.org/grpc/internal/pretty@1.60.0 > google.golang.org/protobuf/encoding/protojson@1.31.0
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > google.golang.org/grpc/reflection@1.60.0 > google.golang.org/grpc/reflection/grpc_reflection_v1alpha@1.60.0 > google.golang.org/grpc@1.60.0 > google.golang.org/grpc/internal/transport@1.60.0 > google.golang.org/grpc/internal/pretty@1.60.0 > github.com/golang/protobuf/jsonpb@1.5.3 > google.golang.org/protobuf/encoding/protojson@1.31.0
  Fixed in: 1.32.0

✗ Medium severity issue found in github.com/hashicorp/yamux
  Description: MPL-2.0 license
  Info: https://snyk.io/vuln/snyk:lic:golang:github.com:hashicorp:yamux:MPL-2.0
  Introduced through: github.com/mattermost/mattermost/server/public/plugin@0.0.12
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > github.com/hashicorp/yamux@0.1.1
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > github.com/hashicorp/go-plugin/internal/grpcmux@1.6.0 > github.com/hashicorp/yamux@0.1.1

✗ Medium severity issue found in github.com/hashicorp/go-plugin
  Description: MPL-2.0 license
  Info: https://snyk.io/vuln/snyk:lic:golang:github.com:hashicorp:go-plugin:MPL-2.0
  Introduced through: github.com/mattermost/mattermost/server/public/plugin@0.0.12
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > github.com/hashicorp/go-plugin/internal/cmdrunner@1.6.0
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/hashicorp/go-plugin@1.6.0 > github.com/hashicorp/go-plugin/internal/grpcmux@1.6.0
  and 3 more...

✗ Medium severity issue found in github.com/go-sql-driver/mysql
  Description: MPL-2.0 license
  Info: https://snyk.io/vuln/snyk:lic:golang:github.com:go-sql-driver:mysql:MPL-2.0
  Introduced through: github.com/mattermost/mattermost/server/public/plugin@0.0.12
  From: github.com/mattermost/mattermost/server/public/plugin@0.0.12 > github.com/go-sql-driver/mysql@1.7.1
```

**Comment by marcos-cereijo-pexip** [comment]
Updated the dependencies and now all the Snyk problems are solved. Could anyone take a look to the PR?

File: Makefile [hunk 1]
```diff
@@ -8,8 +8,11 @@ GO_TEST_FLAGS ?= -race
 GO_BUILD_FLAGS ?=
 MM_UTILITIES_DIR ?= ../mattermost-utilities
 DLV_DEBUG_PORT := 2346
-DEFAULT_GOOS := $(shell go env GOOS)
-DEFAULT_GOARCH := $(shell go env GOARCH)
+# DEFAULT_GOOS := $(shell go env GOOS)
```
Comment (maalfrid) [inline Makefile:11]: It is intended to comment out these 2 lines and keep 
```
DEFAULT_GOOS := "linux"
DEFAULT_GOARCH := "amd64"
```
?

File: webapp/webpack.config.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
-const path = require('path')
+import path from 'path'
 
-module.exports = {
+import { fileURLToPath } from 'url'
+const _filename = fileURLToPath(import.meta.url)
+// eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/naming-convention
+const __dirname = path.dirname(_filename)
```
Comment (maalfrid) [inline webapp/webpack.config.js:6]: Just to understand, what changed so that you have to set the `__dirname` const?

**Review by maalfrid — APPROVED** [review]
Based on my limited knowledge about this project, I'm approving:) You wanna have a look @timwilking / @lzgrzebski before this PR is merged?

**Comment by marcos-cereijo-pexip** [comment]
@maalfrid could you approve this PR?

**Comment by maalfrid** [comment]
@marcos-cereijo-pexip weird that I did approve (twice actually), but merging is blocked even though you have 1 approval🤔
Screenshot 2024-01-17 at 21 05 01

**Pull request merged**